src/EditorFeatures/CSharpTest/ConvertToRecord/ConvertToRecordTests.cs

```diff
@@ -0,0 +1,864 @@
+        }
+
+        [Fact]
+        public async Task TestMoveSimpleProperty()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; init; }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public record C(int P)
+    {
+    }
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMoveMultipleProperties()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; init; }
+        public bool B { get; init; }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public record C(int P, bool B)
+    {
+    }
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMoveMultiplePropertiesOnStruct()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public struct [|C|]
+    {
+        public int P { get; set; }
+        public bool B { get; set; }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public record struct C(int P, bool B)
+    {
+    }
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMoveMultiplePropertiesOnReadonlyStruct()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public readonly struct [|C|]
+    {
+        public int P { get; init; }
+        public bool B { get; init; }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public readonly record struct C(int P, bool B)
+    {
+    }
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMovePropertiesWithAttributes()
+        {
+            var initialMarkup = @"
+using System;
+
+namespace N
+{
+    public class [|C|]
+    {
+        [Obsolete(""P is Obsolete"", error: true)]
+        public int P { get; init; }
+
+        [Obsolete(""B will be obsolete, error: false"")]
+        public bool B { get; init; }
+    }
+}
+";
+            var changedMarkup = @"
+using System;
+
+namespace N
+{
+    public record C([property: Obsolete(""P is Obsolete"", error: true)] int P, [property: Obsolete(""B will be obsolete, error: false"")] bool B)
+    {
+    }
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMovePropertiesAndDeleteSimpleEqualOperators1()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; init; }
+        public bool B { get; init; }
+
+        public static bool operator ==(C c1, object? c2) {
```

Comment:
interesting.  do you have tests of `==` and `!=` that compare two vlaues of the same type (e.g. C not objecT).